src/extension/tab.ts

```diff
@@ -0,0 +1,22 @@
+import { v4 as uuid } from 'uuid'
+import { OnlyRunInContext } from '@holoflows/kit/es'
+import { lastActivatedTabIdSettings } from '../settings/settings'
+import { sideEffect } from '../utils/side-effects'
+import { untilDocumentReady } from '../utils/dom'
+
+OnlyRunInContext(['content', 'options', 'debugging'], 'tab')
+
+export const TAB_ID = uuid() // create a id for every tab
+
+const setId = () => {
+    if (document.visibilityState === 'visible') console.log(`set id to ${TAB_ID}`)
+    if (document.visibilityState === 'visible') lastActivatedTabIdSettings.value = TAB_ID
+}
+
+sideEffect.then(untilDocumentReady).then(() => {
```

Comment:
This design doesn't seem right. Plz open an issue to draft what the problem it want to resolve so we can find a better way later.

Reply:
#1590